packages/ui/src/machines/authenticator/index.ts

```diff
@@ -128,11 +175,17 @@ export function createAuthenticatorMachine() {
       actions: {
         forwardToActor: forwardTo((context) => context.actorRef),
         setUser: assign({
-          user: (_, event) => event.data.user || event.data,
+          user: (_, event) => event.data as CognitoUserAmplify,
         }),
```

Comment:
We're making this implementation more precise -- `setUser` will now only be called by `Auth.*` service calls, which has its information on `event.data`. Any `user` data passed between transitions will be handled more explicitly through `setActorDoneData`.